docs/content/Checkbox.md

```diff
@@ -0,0 +1,104 @@
+---
+title: Checkbox
+status: Alpha
+---
+
+`Checkbox` is a form component that applies Primer design language to a native HTML checkbox input element.
+
+## Default example
+
+The `Checkbox` component can operate in both controlled and uncontrolled modes.
+
+```jsx live
+<>
+  <Box as="form" p={3} pt={0} sx={{display: 'flex', alignItems: 'center'}}>
+    <Checkbox id="default-checkbox" />
+    <Text as="label" htmlFor="default-checkbox" sx={{fontSize: 2, fontWeight: 'bold', marginLeft: 1}}>
+      Default checkbox
+    </Text>
+  </Box>
+  <Box as="form" p={3} sx={{display: 'flex', alignItems: 'center'}}>
+    <Checkbox id="always-checked-checkbox" checked={true} />
+    <Text as="label" htmlFor="always-checked-checkbox" sx={{fontSize: 2, fontWeight: 'bold', marginLeft: 1}}>
+      Always checked
+    </Text>
+  </Box>
+  <Box as="form" p={3} sx={{display: 'flex', alignItems: 'center'}}>
+    <Checkbox id="always-unchecked-checkbox" checked={false} />
+    <Text as="label" htmlFor="always-unchecked-checkbox" sx={{fontSize: 2, fontWeight: 'bold', marginLeft: 1}}>
+      Always unchecked
+    </Text>
+  </Box>
+
+  <Box as="form" p={3} sx={{display: 'flex', alignItems: 'center'}}>
+    <Checkbox id="inactive-checkbox" checked={true} disabled />
+    <Text as="label" htmlFor="inactive-checkbox" sx={{fontSize: 2, fontWeight: 'bold', marginLeft: 1}}>
+      Inactive
+    </Text>
+  </Box>
+</>
+```
+
+import { useState } from 'react'
+import { useMDXScope } from "gatsby-plugin-mdx/context";
+
+export const MyDynamicButton = () => {
+const [count, setCount] = useState(0)
+return (
+<button onClick={() => setCount(count + 1)}>Clicked {count} times!</button>
+)
+}
+
+<Note variant="warning">
+Checkbox components should always be accompanied by a corrosponding label to improve support for assistive technologies.
+</Note>
+
+## Indeterminate example
+
+An `indeterminate` checkbox state can be used if the inputs value is in-between on or off. This can be useful in situations where you are required to display an incomplete state, that is dependent on other external selections to force a binary decision of the parent checkbox.
```

Comment:
Instead of saying "in-between on or off", I'd recommend something like "is neither true nor false"